nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java

```diff
@@ -76,7 +79,12 @@
 
     @Override
     @SuppressWarnings("unchecked")
-    public List<DocumentModel> getComments(CoreSession session, DocumentModel docModel) {
+    public List<DocumentModel> getComments(CoreSession session, DocumentModel docModel)
+            throws CommentSecurityException {
+        if (session.exists(docModel.getRef()) && !session.hasPermission(docModel.getRef(), SecurityConstants.READ)) {
+            throw new CommentSecurityException("The user " + session.getPrincipal().getName()
+                    + " does not have access to the comments of document " + docModel.getId());
+        }
```

Comment:
Shouldn't we retrieve comments with a privileged session ?

Reply:
Yes indeed, my mistake